codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsProtocolUtils.java

```diff
@@ -117,6 +118,15 @@ static void generateJsonParseBody(GenerationContext context) {
         writer.write("");
     }
 
+    static void generateJsonParseBodyWithQueryHeader(GenerationContext context) {
+        TypeScriptWriter writer = context.getWriter();
+        writer.addImport("HeaderBag", "__HeaderBag", "@aws-sdk/types");
+        writer.addImport("HttpResponse", "__HttpResponse", "@aws-sdk/protocol-http");
```

Comment:
The `HttpResponse` and `SerdeContext` imports don't look used by the loaded stub, can they be removed?